posthog/hogql/database/schema/util/where_clause_visitor.py

```diff
@@ -0,0 +1,72 @@
+from posthog.hogql import ast
+
+
+class ASTVisitor:
+    def visit(self, node: ast.Expr) -> ast.Expr:
+        if isinstance(node, ast.And):
+            return self.visit_and(node)
+        elif isinstance(node, ast.Or):
+            return self.visit_or(node)
+        elif isinstance(node, ast.Not):
+            return self.visit_not(node)
+        elif isinstance(node, ast.Call):
+            return self.visit_call(node)
+        elif isinstance(node, ast.Field):
+            return self.visit_field(node)
+        elif isinstance(node, ast.Constant):
+            return self.visit_constant(node)
+        elif isinstance(node, ast.CompareOperation):
+            return self.visit_compare_operation(node)
+        elif isinstance(node, ast.ArithmeticOperation):
+            return self.visit_arithmetric_operation(node)
+        elif isinstance(node, ast.Placeholder):
+            return self.visit_placeholder(node)
+        else:
+            raise Exception(f"Unknown node type {type(node)}")
+
+    def visit_and(self, node: ast.And) -> ast.Expr:
+        return ast.And(exprs=[self.visit(node) for node in node.exprs])
+
+    def visit_or(self, node: ast.Or) -> ast.Expr:
+        return ast.Or(exprs=[self.visit(node) for node in node.exprs])
+
+    def visit_not(self, node: ast.Not) -> ast.Expr:
+        return ast.Not(expr=self.visit(node.expr))
+
+    def visit_call(self, node: ast.Call) -> ast.Expr:
+        return ast.Call(name=node.name, args=[self.visit(arg) for arg in node.args])
+
+    def visit_field(self, node: ast.Field) -> ast.Expr:
+        return node
+
+    def visit_constant(self, node: ast.Constant) -> ast.Expr:
+        return node
+
+    def visit_compare_operation(self, node: ast.CompareOperation) -> ast.Expr:
+        return ast.CompareOperation(
+            op=node.op,
+            left=self.visit(node.left),
+            right=self.visit(node.right),
+        )
+
+    def visit_arithmetric_operation(self, node: ast.ArithmeticOperation) -> ast.Expr:
+        return ast.ArithmeticOperation(
+            op=node.op,
+            left=self.visit(node.left),
+            right=self.visit(node.right),
+        )
+
+    def visit_placeholder(self, node: ast.Placeholder) -> ast.Expr:
+        return node
+
+
+class StripInfoVisitor(ASTVisitor):
+    def visit_field(self, node: ast.Field) -> ast.Expr:
+        return ast.Field(chain=node.chain)
+
+    def visit_constant(self, node: ast.Constant) -> ast.Expr:
+        return ast.Constant(value=node.value)
+
+    @staticmethod
+    def strip_info(node: ast.Expr) -> ast.Expr:
+        return StripInfoVisitor().visit(node)
```

Comment:
I don't exactly know what this does, but `clone_expr` has something similar built in if you enable one or both of the boolean arguments:

```py
def clone_expr(expr: Expr, clear_types=False, clear_locations=False) -> Expr:
    """Clone an expression node."""
    return CloningVisitor(clear_types=clear_types, clear_locations=clear_locations).visit(expr)
```

Reply:
Nice! Yeah, best to always inherit from the ones we have. It makes adding new fields and nodes easier.